xlators/mgmt/glusterd/src/glusterd-volgen.c

```diff
@@ -3794,12 +3797,12 @@ set_afr_pending_xattrs_option(volgen_graph_t *graph,
                 goto out;
             memset(afr_xattrs_list, 0, list_size);
             ptr = afr_xattrs_list;
+            curr = ptr;
             i = 1;
             subvol_index++;
             continue;
         }
-        ptr[strlen(brick->brick_id)] = ',';
-        ptr += strlen(brick->brick_id) + 1;
+        *curr++ = ',';
```

Comment:
even though its simple, a serious set of changes. need little slow review, and more eyes here.